tests/framework/e2e/etcdctl.go

```diff
@@ -39,6 +43,23 @@ func NewEtcdctl(cfg *EtcdProcessClusterConfig, endpoints []string) *EtcdctlV3 {
 	}
 }
 
+func (ctl *EtcdctlV3) WithAuth(userName, password string) (*EtcdctlV3, error) {
```

Comment:
I would prefer to avoid mixing binary and library client in the same module. This hides fact how client interface is implemented, which is important for testing. We want to explicitly know that e2e tests use etcdctl and integration tests are using library. 

The tests you are working on are an exemption to the rule. They want to use client in e2e test. This is ok, however it should be explicit.

Reply:
Done in the next PR revision